CMakeLists.txt

```diff
@@ -730,6 +730,30 @@ if (USE_TORCH)
   set(TORCH_INC_DIR ${TORCH_LOCATION}/include/ ${TORCH_LOCATION}/include/torch/csrc/api/include/ ${CMAKE_BINARY_DIR}/pytorch/src/pytorch/torch/include/torch/csrc/api/include ${TORCH_LOCATION}/.. ${CMAKE_BINARY_DIR}/src)
   set(TORCH_LIB_DIR ${TORCH_LOCATION}/lib/)
 
+  # TORCH VISION
+  message(STATUS "Configuring Pytorch Vision")
+  set(PYTORCH_VISION_COMPLETE ${CMAKE_BINARY_DIR}/CMakeFiles/pytorch_vision-complete)
+  # set(PYTORCH_VISION_COMMIT b68adcf9a9280aef02fc08daed170d74d0892361) # 0.6 & torch 1.5
+  set(PYTORCH_VISION_COMMIT 78ed10cc51067f1a6bac9352831ef37a3f842784) # 0.7 & torch 1.6
+  set(TORCHVISION_LOCATION "${CMAKE_BINARY_DIR}/pytorch_vision/src/pytorch_vision-install")
+  ExternalProject_Add(
+    pytorch_vision
+    PREFIX pytorch_vision
+    INSTALL_DIR ${CMAKE_BINARY_DIR}
+    GIT_REPOSITORY https://github.com/pytorch/vision.git
+    GIT_TAG ${PYTORCH_VISION_COMMIT}
+    GIT_CONFIG advice.detachedHead=false
+    UPDATE_DISCONNECTED 1
+    # CONFIGURE_COMMAND test -f ${PYTORCH_VISION_COMPLETE} && echo Skipping || cmake ../pytorch_vision -DWITH_CUDA=ON
+    CMAKE_ARGS -DWITH_CUDA=${PYTORCH_USE_CUDA} -DCMAKE_PREFIX_PATH=${TORCH_LOCATION} -DCMAKE_INSTALL_PREFIX=${TORCHVISION_LOCATION}
+      "-DCMAKE_CXX_FLAGS=-isystem ${CMAKE_BINARY_DIR}/pytorch/src/pytorch/" "-DCMAKE_CUDA_FLAGS=-isystem ${CMAKE_BINARY_DIR}/pytorch/src/pytorch/"
+    BUILD_COMMAND test -f ${PYTORCH_VISION_COMPLETE} && echo Skipping || make -j6
+  )
+
+  list(APPEND TORCH_INC_DIR "${TORCHVISION_LOCATION}/include/")
+  list(APPEND TORCH_LIB_DIR "${TORCHVISION_LOCATION}/lib/")
+  list(APPEND TORCH_LIB_DEPS torchvision)
```

Comment:
I think they must be added on each COMMON_XXX list at the file bottom.

Also an add_dependencies maybe need in src/CMakelists.txt

Reply:
Torchvision stuff is added to TORCH_XXX variables, so it's added in the COMMON_XXX variables

I did it that way because torchvision is always built with pytorch.

Reply:
My bad I misread the variable name.